### PR TITLE
Fix gocyclo issue

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixes: #158

Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by JohnStrunk** [comment]
CI is broken due to changes in gometalinter. #159 fixes gml by updating to v3

File: pkg/glusterfs/controllerserver.go [hunk 1]
```diff
@@ -65,40 +65,31 @@ type ProvisionerConfig struct {
 
 // ParseCreateVolRequest parse incoming volume create request and fill
 // ProvisionerConfig.
-//nolint gocyclo
 func (cs *ControllerServer) ParseCreateVolRequest(req *csi.CreateVolumeRequest) (*ProvisionerConfig, error) {
 
 	var (
 		reqConf ProvisionerConfig
 		gdReq   api.VolCreateReq
-		err     error
```
Comment (humblec) [inline pkg/glusterfs/controllerserver.go]: Wondering when this `err` was added to this struct!!

File: pkg/glusterfs/controllerserver.go [hunk 2]
```diff
@@ -133,17 +124,23 @@ func validateThinArbiter(req *csi.CreateVolumeRequest) error {
...
 }
 
-func addThinArbiter(req *api.VolCreateReq, thinArbiter string) error {
+func addThinArbiter(volReq *api.VolCreateReq, req *csi.CreateVolumeRequest) error {
+
+	if err := validateThinArbiter(req); err != nil {
+		return err
+	}
 
+	thinArbiter := req.Parameters["arbiterPath"]
```
Comment (humblec) [inline pkg/glusterfs/controllerserver.go]: Can we make this key a constant ?

File: pkg/glusterfs/controllerserver.go [hunk 3]
```diff
@@ -133,17 +124,23 @@ func validateThinArbiter(req *csi.CreateVolumeRequest) error {
...
 	s := strings.Split(thinArbiter, ":")
 	if len(s) != 2 && len(s) != 3 {
 		return fmt.Errorf("thin arbiter brick must be of the form <host>:<brick> or <host>:<brick>:<port>")
 	}
 
-	req.Options = map[string]string{
+	volReq.ReplicaCount = 2
```
Comment (humblec) [inline pkg/glusterfs/controllerserver.go]: Why this is hardcoded to "2" ? it should pick it up from a constant

**Comment by humblec** [comment]
@Madhu-1 Please check the comments

File: pkg/glusterfs/controllerserver.go (hunk 1 above)
Comment (Madhu-1) [reply pkg/glusterfs/controllerserver.go]: err was introduced in https://github.com/gluster/gluster-csi-driver/commit/77697b86799b3c92a65ac27a60b30f5d0cdbcc62 
and grouping was done in https://github.com/gluster/gluster-csi-driver/commit/13d257c9381a7d1f1a677d1b217895570c7264fb

File: pkg/glusterfs/controllerserver.go (hunk 3 above)
Comment (Madhu-1) [reply pkg/glusterfs/controllerserver.go]: currently gd2 supports replica 2 and arbiter 1

File: pkg/glusterfs/controllerserver.go (hunk 1 above)
Comment (humblec) [reply pkg/glusterfs/controllerserver.go]: @Madhu-1 I was talking about pushing that error to the struct! that looks strange and admit that we missed it in review.

File: pkg/glusterfs/controllerserver.go (hunk 2 above)
Comment (Madhu-1) [reply pkg/glusterfs/controllerserver.go]: currently, all the keys we are parsing from SC are not constants https://github.com/gluster/gluster-csi-driver/blob/master/pkg/glusterfs/controllerserver.go#L85

if its required i can do that

**Comment by Madhu-1** [comment]
@JohnStrunk PTAL

**Pull request merged**